File: iOS/Code/Xamarin/bugTrap/bugTrapKit/Controllers/BtNavigationController.cs
using System;
using UIKit;
using System.Threading.Tasks;

namespace bugTrapKit
{
	public class BtNavigationController : UINavigationController
	{
		public BtNavigationController ()
		{
			
		}

		public BtNavigationController (IntPtr handle) : base(handle)
		{

		}


		public override void ViewDidLoad ()
		{
			base.ViewDidLoad();

			if (TrapState.Shared.InActionExtension) {
				NavigationBar.BarTintColor = null;
				NavigationBar.TintColor = Colors.Theme;
				NavigationBar.TitleTextAttributes.ForegroundColor = Colors.Black;
			}
		}

		public Task PresentNewBugDetailsNavController (bool animate)
		{
			var kitStoryboard = UIStoryboard.FromName("bugTrapKit", null);

			var newBugDetailsNavController = kitStoryboard.Instantiate<BtNewBugDetailsNavigationController>();
			if (newBugDetailsNavController != null) {
				return PresentViewControllerAsync(newBugDetailsNavController, animate);
			}
			return null;
		}


		public Task PresentAnnotateImageNavController (bool animate)
		{
			var kitStoryboard = UIStoryboard.FromName("bugTrapKit", null);

			var annotateImageNavController = kitStoryboard.Instantiate<BtAnnotateImageNavigationController>();
			if (annotateImageNavController != null) {
				return PresentViewControllerAsync(annotateImageNavController, animate);
			}
			return null;
		}


		public override bool PrefersStatusBarHidden () => true;

		public override UIStatusBarStyle PreferredStatusBarStyle () => UIStatusBarStyle.LightContent;
	}
}

File: iOS/Code/Swift/bugTrap/bugTrapKit/Controllers/BtNavigationController.swift
//
//  BtNavigationController.swift
//  bugTrap
//
//  Created by Colby L Williams on 11/20/14.
//  Copyright (c) 2014 bugTrap. All rights reserved.
//

import UIKit

class BtNavigationController: UINavigationController {

	override func viewDidLoad() {
		super.viewDidLoad()
		
		if TrapState.Shared.inActionExtension {
			navigationBar.barTintColor = nil
			navigationBar.tintColor = Colors.Theme.uiColor
			navigationBar.titleTextAttributes?.updateValue(Colors.Black.uiColor, forKey: NSForegroundColorAttributeName)
		}
	}

	
//	func presentSettingsNavController (animate: Bool, block: (() -> Void)?) {
//		let kitStoryboard = UIStoryboard(name: "bugTrapKit", bundle: nil)
//		
//		if let settingsNavController = kitStoryboard.instantiateViewControllerWithIdentifier("BtSettingsNavigationController") as? UINavigationController {
//			presentViewController(settingsNavController, animated: animate, completion: block)
//		}
//	}
	
	func presentNewBugDetailsNavController (animate: Bool, block: (() -> Void)?) {
		let kitStoryboard = UIStoryboard(name: "bugTrapKit", bundle: nil)
		
		if let newBugDetailsNavController = kitStoryboard.instantiateViewControllerWithIdentifier("BtNewBugDetailsNavigationController") as? UINavigationController {
			presentViewController(newBugDetailsNavController, animated: animate, completion: block)
		}
	}

	func presentAnnotateImageNavController (animate: Bool, block: (() -> Void)?) {
		let kitStoryboard = UIStoryboard(name: "bugTrapKit", bundle: nil)
	
		if let annotateImageNavController = kitStoryboard.instantiateViewControllerWithIdentifier("BtAnnotateImageNavigationController") as? BtAnnotateImageNavigationController {
			presentViewController(annotateImageNavController, animated: animate, completion: block)
		}
	}
	
	override func prefersStatusBarHidden() -> Bool {
		return true
	}
	override func preferredStatusBarStyle() -> UIStatusBarStyle {
		return .LightContent
	}
}